libmamba/src/core/history.cpp

```diff
@@ -223,6 +224,11 @@ namespace mamba
                     r.link_dists.push_back(x.substr(1));
                 }
             }
+            if ((r.link_dists.size() > 0) || (r.unlink_dists.size() > 0))
+            {
+                r.revision_num = i;
+                i += 1;
```

Comment:
`++i` is more idiomatic, or replace the two lines with `r.revision_num = revision_num++;`